File: home/engine/project/src/main/java/com/genos/overlay/receiver/BootReceiver.java
package com.genos.overlay.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

/**
 * Boot receiver to start GENOS services after device boot
 */
public class BootReceiver extends BroadcastReceiver {
    private static final String TAG = "BootReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        Log.i(TAG, "Received broadcast: " + action);

        if (Intent.ACTION_BOOT_COMPLETED.equals(action) || 
            Intent.ACTION_LOCKED_BOOT_COMPLETED.equals(action)) {
            
            Log.i(TAG, "Device boot completed, starting GENOS services");
            
            // Here you could start your services automatically if needed
            // Intent overlayIntent = new Intent(context, OverlayService.class);
            // context.startService(overlayIntent);
            
            // Intent accessibilityIntent = new Intent(context, InputEmulatorService.class);
            // context.startService(accessibilityIntent);
        }
    }
}

File: app/src/main/java/ai/genos/core/accessibility/BootReceiver.kt
package ai.genos.core.accessibility

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

/**
 * Boot receiver for starting GENOS accessibility service on device boot
 */
class BootReceiver : BroadcastReceiver() {
    
    companion object {
        private const val TAG = "BootReceiver"
    }
    
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        
        Logger.logInfo(TAG, "Received broadcast: $action")
        
        when (action) {
            Intent.ACTION_BOOT_COMPLETED,
            Intent.ACTION_MY_PACKAGE_REPLACED,
            Intent.ACTION_PACKAGE_REPLACED -> {
                // Start the accessibility service after boot or app update
                try {
                    val serviceIntent = Intent(context, GenosForegroundService::class.java)
                    context.startForegroundService(serviceIntent)
                    
                    Logger.logInfo(TAG, "Started GENOS accessibility service after $action")
                    
                } catch (e: Exception) {
                    Logger.logError(TAG, "Failed to start service after $action", e)
                }
            }
        }
    }
}